File: data-structures/lists/c-sharp/list_for_loop.cs
/*
Raspberry Pi Foundation
Developed as part of Ada Computer Science 
Usage licensed under CC BY-NC-SA 4.0

Note: This file is designed to be copied out and compiled on your machine.
In order for it to compile properly you need to ensure that the project name is the same as the "namespace" in this file.
To run this file you need to:
1. Copy the contents
2. Paste them into the C# IDE of your choice (Visual Studio, for example)
3. Change the namespace to match your project (if necessary)
4. Compile the program
5. Run the program
*/

using System;
using System.Collections.Generic;

namespace AdaCodeSamples {
    class SingleLists {
        // The Main method is the entry point for all C# programs
        public static void Main() {
            // Declare and initialise the astronaut list
            List<string> astronauts = new List<string>() {
                "Frank Rubio", "Stephen Bowen",  "Sultan Al Neyadi"
            };

            // Output the astronaut list
            DisplayAstronauts(astronauts);
        }

        // Display the values from the given astronaut list
        public static void DisplayAstronauts(List<string> astronauts) {
            Console.WriteLine("Astronauts:");

            // Repeat for all the values in the list using the index i
            int numAstronauts = astronauts.Count;
            for (int i = 0; i < numAstronauts; i++) {
                Console.WriteLine($"Index {i} contains the value {astronauts[i]}");
            }
            Console.WriteLine($"There are {numAstronauts} astronauts in the list");
        }
    }
}

File: data-structures/lists/visual-basic/list_for_loop.vb
' Raspberry Pi Foundation
' Developed as part of Ada Computer Science
' Usage licensed under CC BY-NC-SA 4.0
' Note: This file is designed to be copied out and compiled on your machine.
' To run this file you need to:
' 1. Copy the contents
' 2. Paste them into the Visual Basic IDE of your choice (Visual Studio, for example)
' 3. Compile the program
' 4. Run the program

Module Program

    ' The Main subroutine is the default entry point for a VB program
    Sub Main()
        ' Declare and initialise the astronaut list
        Dim astronauts As New List(Of String)({
            "Frank Rubio",
            "Stephen Bowen",
            "Sultan Al Neyadi"
        })

        ' Output the astronaut list
        DisplayAstronauts(astronauts)
    End Sub

    ' Display the values from the given astronaut list
    Sub DisplayAstronauts(ByVal astronauts As List(Of String))
        Console.WriteLine("Astronauts:")

        ' Repeat for all the values in the list using the index i
        Dim numAstronauts As Integer = astronauts.Count
        For i = 0 To numAstronauts - 1
            Console.WriteLine($"Index {i} contains the value {astronauts(i)}")
        Next

        Console.WriteLine($"There are {numAstronauts} astronauts in the list")
    End Sub
End Module